js/common.js

```diff
@@ -0,0 +1,85 @@
+
+
+
+const getAllCountriesData = async () => {
+    let isLoading = true;
+    try {
+        loading(isLoading);
+        const result = await fetch(`../FullCountriesData.json`);
+        if (!result.ok) {
+            throw "Failed to fetch data";
+        }
+        const json = await result.json();
+        // console.log(json);
+        return json;
+    }
+    catch (error) {
+        console.log(error);
+    } finally {
+        isLoading = false;
+        loading(isLoading);
+    }
+}
+
+const loading = () => {
+    const loader = document.querySelector('.country-details')
+        ? document.querySelector('.country-details')
+        : document.querySelector('.countries-grid');
+
+    if (loader) {
+        loader.innerHTML = `
+                    <div class="loader">
+                        <div class="spinner">
+                            <i class="fa-regular fa-circle-notch fa-spin icon"></i>
+                        </div>
+                    </div>
+                `;
+    }
+    else {
+        loader.innerHtml = '';
+    }
+}
+
+
+const selectCountry = async (e) => {
+    const data = await getAllCountriesData();
+    const countryName = e.target.closest(".country");
+    const country = countryName.dataset.countryName;
+    const findCountry = data.find(({ name }) => name === country)
+
+    if (!findCountry) return;
+
+    window.location.href = `details.html?country=${country}`;
+}
```

Comment:
You can extract this code into service that exposes methods instead of writing the same logic again and again